staging/src/k8s.io/apiserver/pkg/endpoints/discovery/root/discovery_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package root
+
+import (
+	"net/http"
+	"net/http/httptest"
+
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	apidiscoveryv2beta1 "k8s.io/api/apidiscovery/v2beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	genericfeatures "k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	k8sscheme "k8s.io/client-go/kubernetes/scheme"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+)
+
+var scheme = runtime.NewScheme()
+var codecs = runtimeserializer.NewCodecFactory(scheme)
+var serializer runtime.NegotiatedSerializer
+
+const discoveryPath = "/apis"
+const jsonAccept = "application/json"
+const aggregatedAccept = ";g=apidiscovery.k8s.io;v=v2beta1;as=APIGroupDiscoveryList"
```

Comment:
I was confused that this is only a "partial accept", rather than a full accept entry, maybe try:
```suggestion
const jsonAccept = "application/json"
const aggregatedAccept = jsonAccept + ";g=apidiscovery.k8s.io;v=v2beta1;as=APIGroupDiscoveryList"
```
though if you also copy `application/json` that's fine too.

Are you testing protobuf too? Now I'm curious if this will work for protobuf.